### PR TITLE
Check for postcodes that exist but aren't Scottish on the SPD endpoint

### LINKED ISSUE #388: No way of determining if a postcode is not from Scotland, or just doesn't exist
Thanks very much @cblanc for getting the Scottish postcodes stuff sorted! Woo! 🎉

One thing which is missing in this implementation is the ability to determine whether a postcode passed to the Scottish postcodes endpoint is 404ing because it doesn't exist, or 404ing because it's not in Scotland. Example:

- https://api.postcodes.io/postcodes/SW1A%200AA is fine and exists
- https://api.postcodes.io/scotland/postcodes/SW1A%200AA 404s as if it doesn't exist because it's not in Scotland

It'd be great if the endpoint could return something like "wrong country" or "postcode not in area" in such a scenario.

### PR DESCRIPTION
**Pull request by mashedkeyboard**
This fixes #388 - response would now still be a 404, but have error `Postcode exists but not in SPD` if a postcode is looked up that is valid, but not in the Scottish database.

### CODE REVIEW

File: app/lib/errors.js [hunk 1]
```diff
@@ -68,6 +68,12 @@ class PostcodeNotFoundError extends PostcodesioHttpError {
     super(404, "Postcode not found");
   }
 }
+
+class PostcodeNotScottishError extends PostcodesioHttpError {
```
Comment (cblanc) [inline app/lib/errors.js]: Can we change to PostcodeNotSpdError or something to that effect

File: app/controllers/scottish_postcodes_controller.js [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
 exports.show = (request, response, next) => {
   const { postcode } = request.params;
-  if (!Postcode.isValid(postcode.trim()))
+  if (!Pc.isValid(postcode.trim()))
     return next(new InvalidPostcodeError());
 
   ScottishPostcode.find(postcode, (error, result) => {
     if (error) return next(error);
-    if (!result) return next(new PostcodeNotFoundError());
+    if (!result) {
```
Comment (cblanc) [inline app/controllers/scottish_postcodes_controller.js]: Can `next()` be called twice here? Once at LOC30 and once at LOC21-23

Comment (mashedkeyboard) [reply app/controllers/scottish_postcodes_controller.js]: Good catch - will `return` the value of `Postcode.find` to fix.

File: app/lib/errors.js (hunk 1 above)
Comment (mashedkeyboard) [reply app/lib/errors.js]: Done! :)

File: app/controllers/scottish_postcodes_controller.js (hunk 2 above)
Comment (mashedkeyboard) [reply app/controllers/scottish_postcodes_controller.js]: Fixed I think - I'm not a Node.js expert, but I reckon that's going to work how I think it is.

File: test/scottish_postcodes.integration.js [hunk 3]
```diff
@@ -112,5 +112,23 @@ describe("Scottish postcode route", () => {
           done();
         });
     });
+
+    it("should return error if postcode not found but is found in main database", done => {
+      const postcode = "SW1A0AA";
```
Comment (cblanc) [inline test/scottish_postcodes.integration.js]: Can use `M1 1AD` - a non-Scottish postcode that exists in seed

File: app/lib/errors.js [hunk 4]
```diff
@@ -68,6 +68,12 @@ class PostcodeNotFoundError extends PostcodesioHttpError {
     super(404, "Postcode not found");
   }
 }
+
+class PostcodeNotInSpdError extends PostcodesioHttpError {
+  constructor() {
+    super(404, "Postcode exists but not in SPD");
```
Comment (cblanc) [inline app/lib/errors.js]: Sorry I missed this earlier - "Postcode exists in ONSPD but not in SPD"

**Comment by mashedkeyboard** [comment]
Sorry, didn't manage to get around to finishing this off yesterday. That should be all sorted now for all the requested changes - and tests are passing, woo!

**Pull request merged**